beacon-chain/rpc/eth/validator/handlers.go

```diff
@@ -86,13 +95,15 @@ func (s *Server) SubmitContributionAndProofs(w http.ResponseWriter, r *http.Requ
 		http2.HandleError(w, "No data submitted", http.StatusBadRequest)
 		return
 	}
-
 	var req SubmitContributionAndProofsRequest
 	if err := json.NewDecoder(r.Body).Decode(&req.Data); err != nil {
 		http2.HandleError(w, "Could not decode request body: "+err.Error(), http.StatusBadRequest)
 		return
 	}
-
+	if len(req.Data) == 0 {
+		http2.HandleError(w, "No data submitted", http.StatusBadRequest)
+		return
+	}
 	validate := validator.New()
 	if err := validate.Struct(req); err != nil {
```

Comment:
does this do a deep validate? i.e. do the validation of each individual item inside? if not we need to add it below

Reply:
yes, it does